### ISSUE TITLE
Build dependency in git subfolder

### ISSUE THREAD

**Issue by gmarti** [oxalica/nocargo#3]
Hello,

I'm trying to build a crate from source so in my Cargo.toml I have :
`flatdata = {git = "https://github.com/heremaps/flatdata", rev = "ac1558b1425ffcffb1a81e4f615808689af5c64e" }`

This is the result in Cargo.lock:
```
[[package]]
name = "flatdata"
version = "0.5.4"
source = "git+https://github.com/heremaps/flatdata?rev=ac1558b1425ffcffb1a81e4f615808689af5c64e#ac1558b1425ffcffb1a81e4f615808689af5c64e"
dependencies = [
 "diff",
 "memmap2",
 "tar",
 "walkdir",
]
```

Its supposed to build the flatadata-rs subfolder of the repository but i'm getting :
> error: getting status of '/nix/store/6ij515l5zv5wkfw289czwh68zivxic22-source/Cargo.toml': No such file or directory

I tried to override the `src` of the crate (with or without the #):
```
ws = nocargo.lib.${system}.mkRustPackageOrWorkspace {
  src = ./.;

  gitSrcs = {
    "https://github.com/heremaps/flatdata?rev=ac1558b1425ffcffb1a81e4f615808689af5c64e" = inputs.flatdata-src;
  };

  buildCrateOverrides = with nixpkgs.legacyPackages.${system}; {
    "flatdata 0.5.4 (git+https://github.com/heremaps/flatdata?rev=ac1558b1425ffcffb1a81e4f615808689af5c64e)" = old: {
      src = "${old.src}/flatdata-rs";
    };

  };
};
```

But it doesn't seem to work, any thoughts ?

Thanks for the help

**Comment by oxalica** [comment]
Git dependency containing workspace are not supported now.

As a workaround, you can point `gitSrcs` to the directories of individual packages.
```
gitSrcs = {
  "https://github.com/heremaps/flatdata?rev=ac1558b1425ffcffb1a81e4f615808689af5c64e" = inputs.git-1 + "/flatdata-rs/lib";
};
```

**Comment by gmarti** [comment]
Thx :+1: 

I'm no blocked further
Seems the flatadata dep is not available in build.rs

Cargo.toml
```
[dependencies]
flatdata = {git = "https://github.com/heremaps/flatdata", rev = "ac1558b1425ffcffb1a81e4f615808689af5c64e", features = ["tar"] }

[build-dependencies]
flatdata = {git = "https://github.com/heremaps/flatdata", rev = "ac1558b1425ffcffb1a81e4f615808689af5c64e", features = ["tar"] }
```

build.rs
```
fn main() {
    flatdata::generate(
        "file.flatdata",
        &std::env::var("OUT_DIR").unwrap(),
    )
    .expect("generator failed");
}
```

**Comment by gmarti** [comment]
Seems related to https://github.com/oxalica/nocargo/issues/7

**Comment by oxalica** [comment]
Fixed in 67053a81af61a45346e6c2edb520cc38d994c90f

**Comment by gmarti** [comment]
Works like a charm :+1: thank you very much
